lib/tools/apk-utils.js

```diff
@@ -474,12 +474,81 @@ apkUtilsMethods.getDeviceLocale = async function () {
 
 /**
  * Set the locale name of the device under test.
+ * Compatibility for appium-android-driver 1.32.4-
  *
  * @param {string} locale - The name of the new device country.
  */
 apkUtilsMethods.setDeviceLocale = async function (locale) {
-  // this method is only used in API >= 23
-  await this.setDeviceSysLocale(locale);
+  let split_locale = locale.split("-");
+  await this.setDeviceLanguageCountry(split_locale[0], split_locale[1]);
+};
+
+/**
+ * Set the locale name of the device under test.
+ *
+ * @param {string} language - Language. e.g. en, ja
+ * @param {string} country - Country. e.g. US, JP
+ */
+apkUtilsMethods.setDeviceLanguageCountry = async function (language, country) {
```

Comment:
- in `lib/tools/adb-commands.js`, focus on calling adb commands
- in `lib/tools/apk-utils.js`, handle adb commands